### PR TITLE
feat: chapter-level resolution, docs, and final fixes (#64)

### LINKED ISSUE #62: refactor: enrichment plugin architecture (registry/pipeline pattern)
## Summary

Refactor OpenAlex and Open Library enrichment into a registry/pipeline pattern so enrichers can be composed, ordered, and selectively enabled.

## Motivation

Currently enrichment logic is scattered: OpenAlex enrichment lives in `enrich_paper`, Open Library enrichment is called inline in `_book_enrichment.py`, and both are wired manually in `_server_tools.py`. As more enrichers are added (Google Books, CrossRef, WorldCat, etc.) this approach will become hard to maintain.

## Proposed work

- Define an `Enricher` protocol (or abstract base class) with a `can_enrich(paper) -> bool` and `enrich(paper, bundle) -> None` interface
- Implement `OpenAlexEnricher` and `OpenLibraryEnricher` as concrete enrichers
- Register enrichers in a list on `ServiceBundle` (or a dedicated `EnrichmentPipeline` object)
- Tools call `pipeline.enrich(papers)` instead of calling enrichers directly
- Enrichers can be enabled/disabled via config or at registration time (e.g. patent tools don't need book enrichment)

## Notes

- Should be a pure refactor — no user-visible behavior changes
- Enricher ordering matters (e.g. OpenAlex before Open Library if OA data informs ISBN lookup)
- Consider a `tags` field on each enricher for selective execution (`enrich_paper` with `fields=["affiliations"]` should only run OpenAlex, not Open Library)

### PR DESCRIPTION
**Pull request by pvliesdonk**
## Summary

- Add `BookChapterRecord` TypedDict and `_chapter_parser.py` with regex-based citation string parsing (chapter numbers, page ranges, parent titles, ISBNs)
- Integrate chapter parsing into `batch_resolve` (paper results get `chapter_info` when citation strings contain chapter patterns) and patent NPL resolution
- CrossRef `book-chapter` metadata takes precedence over regex-parsed hints
- Wire `GoogleBooksEnricher` into book tool resolve functions so it actually runs
- Fix `worldcat_url` gap in `_resolve_work`
- Update all documentation: README.md, docs/configuration.md, docs/tools/index.md, .claude-plugin/plugin/README.md

Closes #62, #67, #61, #66, #68, #64 (completes v0.6.0 milestone).

## Chapter resolution flow

1. `parse_chapter_hint(citation_string)` extracts `ChapterHint` (chapter_number, page_start, page_end, parent_title, isbn)
2. If CrossRef enrichment already identified `type: "book-chapter"`, its structured data is used (`citation_source: "crossref"`)
3. Otherwise, parsed regex hints are used (`citation_source: "parsed"`)
4. `chapter_info` dict is attached to paper results in `batch_resolve` and NPL refs in `get_patent` citations

## New files

| File | Purpose |
|------|---------|
| `_chapter_parser.py` | `ChapterHint` dataclass, `parse_chapter_hint()`, `hint_to_dict()` |

## Test plan

- [x] 19 tests for chapter parser (patterns, edge cases, hint_to_dict)
- [x] 3 tests for batch_resolve chapter integration (parsed, CrossRef precedence, no-chapter)
- [x] 3 tests for patent NPL chapter integration (with S2, without S2, no-chapter)
- [x] Documentation updated for all v0.6.0 features (28 tools, new config, new fields)
- [x] All 778 tests pass, mypy clean, ruff clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/scholar_mcp/_chapter_parser.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    """
+    info: dict[str, Any] = {"citation_source": "parsed"}
+    if hint.chapter_number is not None:
+        info["chapter_number"] = hint.chapter_number
+    if hint.page_start is not None:
+        info["page_start"] = hint.page_start
+    if hint.page_end is not None:
+        info["page_end"] = hint.page_end
+    if hint.parent_title is not None:
+        info["chapter_title"] = hint.parent_title
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_chapter_parser.py]: Fixed in 2768a0d. `hint_to_dict` now emits `parent_title` for the containing-book title, and the `BookChapterRecord` TypedDict gained a `parent_title: str` field.

File: src/scholar_mcp/_tools_utility.py [hunk 2]
```diff
@@ -22,6 +23,47 @@
...
+
+    Args:
+        result: Mutable paper result dict to enrich in-place.
+        raw: Raw identifier string potentially containing chapter hints.
+        paper_data: S2 or OpenAlex paper data dict.
+    """
+    hint = parse_chapter_hint(raw)
+    if not hint.has_chapter_info:
+        return
+    crossref_meta = paper_data.get("crossref_metadata", {})
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_tools_utility.py]: Fixed in 2768a0d. Dropped the CrossRef precedence branch entirely — `_attach_chapter_info` now records parsed hints only, and the docstring calls out that the enrichment pipeline resolves the parent book separately. The function dropped its unused `crossref_meta` parameter and both call sites were updated.

File: src/scholar_mcp/_record_types.py [hunk 3]
```diff
@@ -32,6 +32,22 @@ class BookRecord(TypedDict, total=False):
...
+    """Chapter-level metadata within a book.
+
+    Used when citation strings reference specific chapters or page
+    ranges. ``citation_source`` indicates whether data came from
+    CrossRef (structured) or regex parsing (heuristic).
+    """
+
+    chapter_title: str
+    chapter_number: int
+    page_start: int
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_record_types.py:45]: Fixed in 2768a0d. `citation_source` is now typed as `Literal["crossref", "parsed"]` in `_record_types.BookChapterRecord`.

File: src/scholar_mcp/_chapter_parser.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+def hint_to_dict(hint: ChapterHint) -> dict[str, Any]:
+    """Convert a ChapterHint to a chapter_info dict for JSON output.
+
+    Args:
+        hint: Parsed chapter hint.
+
+    Returns:
+        Dict with populated fields and ``citation_source`` set to ``"parsed"``.
+    """
+    info: dict[str, Any] = {"citation_source": "parsed"}
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_chapter_parser.py:171]: Fixed in 2768a0d. `hint_to_dict` now emits `isbn` when present, and `BookChapterRecord` gained an `isbn: str` field. Covered by a new test (`test_hint_to_dict_includes_isbn_when_present`).

File: docs/tools/index.md [hunk 5]
```diff
@@ -468,13 +508,13 @@ Sections are fetched concurrently where possible (cache lookups run in parallel;
     ],
     "npl_refs": [
       {"raw": "Smith et al., \"Widget Processing\", 2018, doi:10.1234/test", "paper": {"paperId": "abc123", "title": "Widget Processing"}, "confidence": "high"},
-      {"raw": "Doe, \"Advanced Widgets\", 2019", "confidence": null}
+      {"raw": "Doe, \"Advanced Widgets\", Ch. 5, pp. 112-130, 2019", "confidence": null, "chapter_info": {"chapter": "5", "pages": "112-130"}}
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply docs/tools/index.md]: Fixed in 2768a0d. Updated the example to match `hint_to_dict` output: `{"citation_source": "parsed", "chapter_number": 5, "page_start": 112, "page_end": 130}`.

File: src/scholar_mcp/_tools_books.py [hunk 6]
```diff
@@ -340,6 +340,7 @@ async def _resolve_isbn(isbn: str, bundle: ServiceBundle) -> str:
 
     book: BookRecord = normalize_book(edition, source="edition")
     await enrich_authors_from_work(book, bundle)
+    await bundle.enrichment.enrich([book], bundle, tags={"books"})  # type: ignore[list-item]
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_tools_books.py]: Fixed in 2768a0d. `EnrichmentPipeline.enrich()` now takes `Sequence[EnrichableRecord]`, where `EnrichableRecord` is a TypeAlias union of `dict[str, Any] | BookRecord | BookChapterRecord | StandardRecord`. `Sequence` is covariant, so `list[BookRecord]` is now accepted statically. All three `# type: ignore[list-item]` suppressions are gone and mypy is clean.

File: src/scholar_mcp/_chapter_parser.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_chapter_parser.py]: Fixed in 2768a0d. `hint_to_dict` now emits `parent_title` (matching the `ChapterHint` attribute name and the semantic meaning), and `BookChapterRecord` gained a `parent_title: str` field.

File: src/scholar_mcp/_tools_utility.py [hunk 7]
```diff
@@ -22,6 +23,47 @@
...
+            "citation_source": "crossref",
+        }
+        if crossref_meta.get("page"):
+            parts = crossref_meta["page"].split("-")
+            try:
+                chapter_info["page_start"] = int(parts[0])
+                if len(parts) > 1:
+                    chapter_info["page_end"] = int(parts[1])
+            except ValueError:
+                pass
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/scholar_mcp/_tools_utility.py]: Not applicable after 2768a0d — the CrossRef precedence branch in `_attach_chapter_info` was dead code (S2 responses never carry `crossref_metadata` at this point) and has been removed entirely. CrossRef handling lives in the enrichment pipeline (`_enricher_crossref`), which parses the `page` field via dedicated logic rather than a naive `split("-")`.

**Comment by pvliesdonk** [comment]
## Rebased onto `main` and addressed all review comments

This PR was rebased from `feat/enrichment-data-sources` (already squash-merged as #118) onto `main` using `git rebase --onto origin/main <old-base>`. The base has been retargeted to `main` via `gh pr edit`.

### Review comments addressed (commit `2768a0d`)

| # | Reviewer | File | Fix |
|---|---|---|---|
| 3069693230 | claude-bot | `_chapter_parser.py:179` | Rename `chapter_title` → `parent_title` in `hint_to_dict` |
| 3069694445 | gemini-bot | `_chapter_parser.py:179` | (same as above) |
| 3069693487 | claude-bot | `_chapter_parser.py:171` | `hint_to_dict` now emits `isbn` when present |
| 3069693326 | claude-bot | `_tools_utility.py:42` | Drop dead CrossRef branch in `_attach_chapter_info` |
| 3069694450 | gemini-bot | `_tools_utility.py:54` | N/A: branch removed (CrossRef handled by `_enricher_crossref`) |
| 3069693445 | claude-bot | `_record_types.py:45` | `citation_source: Literal["crossref", "parsed"]` |
| 3069693599 | claude-bot | `docs/tools/index.md:511` | Fix NPL `chapter_info` example to match `hint_to_dict` schema |
| 3069693681 | claude-bot | `_tools_books.py:343,433,456` | Broaden `EnrichmentPipeline.enrich()` to `Sequence[EnrichableRecord]`; remove all three `# type: ignore[list-item]` |

### New/updated types

- `BookChapterRecord` gained `parent_title: str` and `isbn: str` fields.
- `_enrichment.EnrichableRecord`: new `TypeAlias` union (`dict[str, Any] | BookRecord | BookChapterRecord | StandardRecord`); `Sequence` provides covariance so any `list[TypedDictSubtype]` is accepted.

### Test / docs delta

- `test_chapter_parser`: updated `test_hint_to_dict` keys; added `test_hint_to_dict_includes_isbn_when_present`.
- `test_tools_utility`: replaced `test_batch_resolve_chapter_crossref_precedence` with `test_batch_resolve_chapter_parent_title_and_isbn` (asserts parsed `parent_title` + `isbn` flow through).
- `test_tools_patent`: `chapter_title` → `parent_title` in the NPL assertion.
- `_chapter_parser._RE_PARENT` now stops at `, ISBN` in addition to `, <year>`, so citations like `In: Deep Learning, ISBN: 978-…` don't swallow the ISBN tail.
- `docs/tools/index.md`: example JSON updated.

### Local gates (all green)

- `uv run ruff check .` — All checks passed!
- `uv run ruff format --check .` — 90 files already formatted
- `uv run mypy src/` — Success: no issues found in 43 source files
- `uv run pytest -x -q` — 778 passed

Ready for re-review.

File: docs/tools/index.md [hunk 8]
```diff
@@ -377,7 +417,7 @@ Resolve up to 100 paper, patent, or book identifiers to full metadata in a singl
 
 **Returns:** JSON list of resolved items:
 
-- **Paper results** have a `"paper"` key. Papers not found in Semantic Scholar are automatically tried via OpenAlex (by DOI); results from OpenAlex include `"source": "openalex"`.
+- **Paper results** have a `"paper"` key. Papers not found in Semantic Scholar are automatically tried via OpenAlex (by DOI); results from OpenAlex include `"source": "openalex"`. When the citation string contains chapter patterns (e.g. "Chapter 3", "pp. 45-67"), a `chapter_info` dict is attached with parsed chapter/page information. CrossRef metadata takes precedence over parsed hints.
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply docs/tools/index.md]: Fixed in 0d151aa. Removed the "CrossRef metadata takes precedence over parsed hints." sentence from the `batch_resolve` paper-result description.
